client/lib/pages/add-plant.dart

```diff
@@ -34,147 +128,290 @@ class _AddPlant extends State<AddPlant> {
+                        },
+                      ),
+                      AddPlantTextField(
+                        label: 'Inhoud pot',
+                        keyboardType: TextInputType.number,
+                        validator: (String value) {
+                          double temp = double.tryParse(value);
 
-                              return null;
-                            },
-                            onSaved: (String value) {
-                              newPlant.potVolume = double.parse(value);
-                            },
-                        ),
-                        Row(
-                          children: <Widget>[
-                            Text('Datum laatste keer water'),
-                            IconButton(
-                              icon: Icon(Icons.date_range, color: Colors.blue,),
-                              onPressed: () => DatePicker.showDateTimePicker(
-                                context,
-                                minTime: DateTime(DateTime.now().year, 1, 1),
-                                maxTime: DateTime.now(),
-                                onConfirm: (DateTime date) {
-                                  newPlant.lastWaterDate = date;
-                                }
-                              ),
+                          if (temp == null) {
+                            return 'Moet een getal zijn.';
+                          }
+
+                          return null;
+                        },
+                        onSaved: (String value) {
+                          newPlant.potVolume = double.parse(value);
+                        },
+                      ),
+                      Row(
+                        children: <Widget>[
+                          Expanded(
+                            flex: 6,
+                            child: Text(
+                                'Wanneer heeft de plant voor het laatst water gekregen?'),
+                          ),
+                          Expanded(
+                            flex: 4,
+                            child: Row(
+                              children: <Widget>[
+                                Text("Weet ik niet"),
+                                Checkbox(
+                                  value: hideDatePicker,
+                                  onChanged: (bool value) {
+                                    print(value);
+                                    hideDatePicker = value;
+                                    // FIXME: find a more maintainable way to enable/disable the submit button
+                                    if (hideDatePicker) {
+                                      allowedToSubmit = true;
+                                    }
+                                    else {
+                                      allowedToSubmit = false;
+                                    }
+                                    setState(() {});
```

Comment:
I think it would be better if the checkbox is in front of the text not after it. Also the text of the checkbox is a bit weird. Maybe change it to `Plant heeft nog geen water gekregen`. That would give the user a better understanding of what the button does.